src/MemCached.php

```diff
@@ -95,40 +92,43 @@ class MemCached extends SimpleCache
     /**
      * @var array list of memcached server configurations
      */
-    private $_servers = [];
-
+    private $_servers;
 
     /**
-     * Initializes this application component.
-     * It creates the memcached instance and adds memcached servers.
+     * @param null $serializer
+     * @param array $servers
+     * @throws InvalidConfigException
+     * @see setSerializer
      */
-    public function init()
+    public function __construct(?SerializerInterface $serializer = null, array $servers = [])
     {
-        parent::init();
-        $this->addServers($this->getMemcached(), $this->getServers());
+        parent::__construct($serializer);
+
+        if (empty($servers)) {
+            $servers = [new MemCachedServer('127.0.0.1')];
+        }
+
+        $this->_servers = $servers;
+
+        $this->addServers($this->getMemcached(), $this->_servers);
```

Comment:
Not good as the $options, $username, $password properties are still not set.

Same old question: how do we manage a constructor that should take a lot of parameters without Yii::buildObject() .. Is it worth having yiisoft/objects with some objects goodies?

Reply:
Is that really a lot of parameters?

Reply:
4 properties in MemCached:
  * $persistentId
  * $options
  * $username
  * $password

2 properties at SimpleCache level:
  * $defaultTtl
  * $keyPrefix

and $serializer & $servers that are already in constructor.

It's a lot.

Reply:
Does not look too bad to me...

Reply:
Let's deal with this later.

If this PR looks ok to you as is, I'll merge it tomorrow and take care of updating references.